### PR TITLE
javascript: : in pairs as punctuation.delimiter

### PR DESCRIPTION
**Pull request by rockerBOO**
Set : to be operator in javascript.

punctuation.delimiter

[old-operator-example](https://user-images.githubusercontent.com/15027/99914664-e6ad4e00-2ccc-11eb-8640-df957f6d3a62.png)

### CODE REVIEW

**Comment by kyazdani42** [comment]
in rust we do not highlight this, in lua we highlight `=` as operator in a table. Is this really a punctuation delimiter or a assignment character ^^ ?

**Comment by rockerBOO** [comment]
mmm I like operator better here. I was replicating the matching from typescript for object type definition `type X {x: Type}` and type delimiter `x: MyType`. I still like operator but should all these  match or are they different?

**Comment by rockerBOO** [comment]
Changed `:` to be an operator.

**Comment by steelsojka** [comment]
I don't think I would consider this an operator at all. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Table 

If you want a it colored a specific way I would extend the javascript query and give a custom highlight group. I think this might add confusion for some colorschemes.

**Comment by rockerBOO** [comment]
https://tc39.es/ecma262/#sec-punctuators
This has `:` in the list of punctuators. I think returning this back to `(pair ":" @punctuation.delimiter` would be appropriate.

**Comment by kyazdani42** [comment]
well depends, but for me `:` in an object is an operator for variable assignment kind of. Just to keep consistency between language, but i wouldn't mind highlighting as a punctuation delimiter (although `:` is not a delimiter, `;` is, or maybe my english is just too limited in this area to understand properly :) )

**Comment by rockerBOO** [comment]
https://tc39.es/ecma262/#sec-object-initializer

This is the object initializer spec. This spec has these punctuations `{` `}` `,` `:`. 
`{`, `}` are `punctuation.bracket`. 
`,` is `punctuation.delimiter`

> [delimiter](https://www.merriam-webster.com/dictionary/delimiter): a character that marks the beginning or end of a unit of data

Using this definition, the delimiter of a key: value is `:`. 

I feel now that `punctuation.delimiter` matches throughout and matches the spec.

**Comment by rockerBOO** [comment]
To further confuse things Atom sets it as a operator. (If I am reading properly)

https://github.com/atom/language-javascript/blob/master/grammars/javascript.cson#L386

File: queries/javascript/highlights.scm [hunk 1]
```diff
@@ -49,6 +49,8 @@
   key: (property_identifier) @method
   value: (arrow_function))
 
+(pair ":" @punctuation.delimiter)
```
Comment (steelsojka) [inline queries/javascript/highlights.scm:52]: Can you move this to the section with the other punctuation?

**Review by steelsojka — CHANGES_REQUESTED** [review]

**Comment by rockerBOO** [comment]
Updated the positioning of this query.

**Comment by steelsojka** [comment]
> To further confuse things Atom sets it as a operator. (If I am reading properly)
> 
> https://github.com/atom/language-javascript/blob/master/grammars/javascript.cson#L386

I tried to look in the text mate grammar for vscode's highlighting to try and compare and I couldn't navigate it very well. If you find another case this is tagged as an operator, we can just change it in another PR. I'm good this for now if you are.

**Review by steelsojka — APPROVED** [review]

**Pull request merged**